client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx

```diff
@@ -11,9 +28,92 @@ type StepRouteProps = {
 	renderStep: ( step: StepperStep ) => JSX.Element | null;
 };
 
+//TODO: Check we can move RenderStep function to here and remove the renderStep prop
 const StepRoute = ( { step, flow, showWooLogo, renderStep }: StepRouteProps ) => {
+	const userIsLoggedIn = useSelector( isUserLoggedIn );
+	const loginUrl = useLoginUrlForFlow( { flow } );
```

Comment:
I think the login URL should bring the user back to the exact step, not only to the flow. Because this now supports logging in at the nth step (not strictly the first).

Reply:
I had this pushing in the step to start with, but I then changed the implementation inside `useLoginUrlForFlow()` to use whatever the current location is (as returned from `useLocation()`), rather than trying to derive or reconstruct the URL. I'd be happy to change it back, but it felt likely to be more solid/reliable if we started from the current URL. (It may be better to use `window.location.pathname + window.location.search` as a starting point, but the result should be similar.)

Reply:
Oh that's certainly less maintenance.